test/unit/krane/kubernetes_resource/daemon_set_test.rb

```diff
@@ -107,6 +107,26 @@ def test_deploy_fails_when_not_all_pods_ready
     refute_predicate(ds, :deploy_succeeded?)
   end
 
+  def test_deploy_waits_for_daemonset_status_to_converge_to_pod_states
+    status = {
+      "desiredNumberScheduled": 1,
+      "updatedNumberScheduled": 1,
+      "numberReady": 0,
+    }
+    not_ready_ds_template = build_ds_template(filename: 'daemon_set.yml', status: status)
+    ready_pod_template = load_fixtures(filenames: ['daemon_set_pods.yml']).first # should be a pod in `Ready` state
+    node_templates = load_fixtures(filenames: ['nodes.yml'])
+    ds = build_synced_ds(ds_template: not_ready_ds_template, pod_templates: [ready_pod_template],
+      node_templates: node_templates)
+    refute_predicate(ds, :deploy_succeeded?)
+
+    status[:numberReady] = 1
+    ready_ds_template = build_ds_template(filename: 'daemon_set.yml', status: status)
```

Comment:
you are basically building a new daemonset here, but i think the behavior is better tested if you take the approach from https://github.com/Shopify/kubernetes-deploy/blob/218a82c3a9b8cad2d3594c1e38ab48af673ee18c/test/unit/krane/kubernetes_resource/daemon_set_test.rb#L58, and just stub and forcefully resync the original one (see https://github.com/Shopify/kubernetes-deploy/blob/218a82c3a9b8cad2d3594c1e38ab48af673ee18c/test/unit/krane/kubernetes_resource/daemon_set_test.rb#L78-L80)